### PR TITLE
Linux installer: small improvements

### PR DESCRIPTION
**Pull request by sklemer1**
- silence zenity
- usable without dbus
- correct shebang
- detect python3 (with some `from __future__` we could make it work with
python2 but is this necessary?)

This is *not* tested very thorowly.

### CODE REVIEW

**Comment by restena-sw** [comment]
You've sent the pull request against master. Do you want this to be part of the 2.0.x series? Then it should go into branch release_2_0 instead.

TBH, when I read "not tested very thoroughly" maybe it's better to merge it by a future 2.0.1 time and not during our short beta phase before going to prod...?

**Comment by sklemer1** [comment]
That's why it's not against release_2_0 but master. I can also make it to apply against 2.0 -- the only difference should be the #114 shebang-line. I don't know how and where to test this without a large audience and many different distros. On my 2 test distros (Debian and Ubuntu) and some mockup scenarios it was working fine. I don't know how much this was tested before as the missing shebang was a real bummer....

**Comment by restena-sw** [comment]
These things are rather thoroguhly tested by @twoln - he has many distros on his list. But then he is so knowledgable that he probably always typed in "python ..." and didn't realise the shebang issue.
Anyway, the content of this pull request is for him in entirety; it's all his code.

File: devices/linux/Files/main.py [hunk 1]
```diff
@@ -303,11 +302,16 @@ def __get_username_password(self):
...
+        
+        # zenity always throws a GTK error to stderr; we shouldn't show it if not in debug mode
+        self.stderr_redir=None
         if os.environ.get('DISPLAY') != None:
             q = subprocess.Popen(['which', 'zenity'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             q.wait()
             if q.returncode == 0:
                 self.graphics = 'zenity'
+                if not debug_on:
+                    self.stderr_redir=subprocess.DEVNULL
```
[the remark this answers is not in the document]
Comment (sklemer1) [reply devices/linux/Files/main.py]: Whoops. Good catch, thank you.

File: devices/linux/Files/main.py [hunk 2]
```diff
@@ -60,18 +46,33 @@ def debug(msg):
     
 def run_installer():
     global debug_on
+    if sys.version_info.major < 3:
```
[the remark this answers is not in the document]
Comment (sklemer1) [reply devices/linux/Files/main.py]: If you use features from later versions, definitely...

File: devices/linux/Files/main.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sklemer1) [reply devices/linux/Files/main.py]: It's easy to change it for py2-support if we decide to want it -- we could just open /dev/null directly or pipe it somewhere else.

**Comment by twoln** [comment]
I think all of these have been implemented. OK to close?

**Pull request closed without merging**